core/src/tile/tileManager.cpp

```diff
@@ -14,16 +14,18 @@
 
 namespace Tangram {
 
-TileManager::TileManager(TileTaskQueue& _tileWorker) : m_workers(_tileWorker) {
+TileManager::TileManager(std::shared_ptr<TileTaskQueue> _tileWorker) : m_workers(_tileWorker) {
 
     m_tileCache = std::unique_ptr<TileCache>(new TileCache(DEFAULT_CACHE_SIZE));
 
     m_dataCallback = TileTaskCb{[this](std::shared_ptr<TileTask>&& task) {
+        if (m_workers) {
             if (task->hasData()) {
-                m_workers.enqueue(std::move(task));
+                m_workers->enqueue(std::move(task));
             } else {
                 task->cancel();
             }
```

Comment:
If the TileManger is already destructed the ```this->m_workers``` reference is an invalid access.

```
 m_dataCallback = TileTaskCb{[m_workers](std::shared_ptr<TileTask>&& task) {
```
should be correct